Auth0/MyAccount/AuthenticationMethods/Auth0MyAccountAuthenticationMethods.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+
+struct Auth0MyAccountAuthenticationMethods: MyAccountAuthenticationMethods {
+
+    let url: URL
+    let session: URLSession
+    let token: String
+
+    var telemetry: Telemetry
+    var logger: Logger?
+
+    init(token: String,
+         url: URL,
+         session: URLSession = .shared,
+         telemetry: Telemetry = Telemetry(),
+         logger: Logger? = nil) {
+        self.token = token
+        self.url = url.appending("authentication-methods")
+        self.session = session
+        self.telemetry = telemetry
+        self.logger = logger
+    }
+
+    // MARK: - Passkey Enrollment
+
+    #if PASSKEYS_PLATFORM
+    @available(iOS 16.6, macOS 13.5, visionOS 1.0, *)
```

Comment:
This is the implementation of the method that requests a passkey enrolment challenge.